### PR TITLE
activity: bug fixes

### PR DESCRIPTION
**Pull request by arthyn**
**frontend**
- [X] Refactor unreads to all flow through same store
- [X] Simplify unread state to string options instead of juggling booleans
- [X] Adjust ID usage to reflect changes from backend
- [X] Change reply line color back to white in non-notify cases
- [X] Fix bug where sometimes navigating away doesn't clear current unread
- [X] Fix issue with group sidebar item not showing dot when children are unread
- [X] Fix issue with Messages sidebar not showing correct items
- [X] Groups are hanging on to unreads for some reason

**backend**
- [X] Proper JSON conversions for stream query
- [X] Adjust children handling so we don't double count
- [X] Fix bug where floor isn't updating in some cases
- [X] Simplify iteration logic when summarizing unreads
Change message key ID time for channels so it reflects the time that - [X] mobile app is interested in and changed thread source to use same ID
- [X] Groups are hanging on to unreads for some reason

Fixes TLON-1974
Fixes TLON-1975

PR Checklist
- [X] Includes changes to desk files
- [ ] Describes how you tested the PR locally (test ship vs livenet)
- [ ] If a new feature, includes automated tests
- [X] Comments added anywhere logic may be confusing without context

### CODE REVIEW

File: desk/app/activity.hoon [hunk 1]
```diff
@@ -279,10 +281,10 @@
   ::  /all: unified feed (equality of opportunity)
   ::
       [%x %all ~]
-    ``activity-stream+!>((tap:on-event:a stream:base))
+    ``activity-stream+!>((gas:on-event:a *stream:a (tap:on-event:a stream:base)))
```
Comment (Fang-) [inline desk/app/activity.hoon:284]:
```suggestion
    ``activity-stream+!>(stream:base)
```

**Review by Fang- — COMMENTED** [review]
Some nitpicks, looks good otherwise.

**Review by Fang- — APPROVED** [review]
Changes look good, thanks!

**Review by latter-bolden — APPROVED** [review]
In terms of functional review, this is feeling great! 

Noticed a few quirks with sidebar dots (communicated out of band), but nothing blocking.

File: apps/tlon-web/src/dms/MessagesSidebarItem.tsx [hunk 2]
```diff
@@ -201,6 +201,7 @@ function MessagesSidebarItem({ whom, pending }: MessagesSidebarItemProps) {
     }
   }, [action, isMobile]);
 
+  console.log(whom);
```
Comment (patosullivan) [inline apps/tlon-web/src/dms/MessagesSidebarItem.tsx:204]:
```suggestion
```

**Review by patosullivan — APPROVED** [review]
Overall looks good, and brian's task in linear covers the other issues I've found (and I'm adding some comments with others).

**Pull request merged**